File: python/scrabble-score/scrabble_score.py
score_table = {
    "AEIOULNRST" : 1,
    "DG"         : 2,
    "BCMP"       : 3,
    "FHVWY"      : 4,
    "K"          : 5,
    "JX"         : 8,
    "QZ"         : 10
}


def score(word):
    return sum(value 
                for ch in word.upper() 
                for key, value in score_table.items() 
                if ch in key)

File: javascript/scrabble-score/scrabble-score.js
const scoreTable = {
  1: ['a', 'e', 'i', 'o', 'u', 'l', 'n', 'r', 's', 't'],
  2: ['d', 'g'],
  3: ['b', 'c', 'm', 'p'],
  4: ['f', 'h', 'v', 'w', 'y'],
  5: ['k'],
  8: ['j', 'x'],
  10: ['q', 'z']
};

const getScore = (char) => {
  for (let value in scoreTable)
    if (scoreTable[value].includes(char)) return Number(value);
}

export const score = (word) =>
  [...word]
    .map(char => getScore(char.toLowerCase()))
    .reduce((sum, value) => { return sum + value }, 0);